app/javascript/components/Waybill/WaybillTable.tsx

```diff
@@ -9,20 +9,24 @@ import { waybillTableCell } from '../../constants/waybillFields';
 import { StyledTableCell, StyledTableRow } from '../../utils/style';
 import httpClient from '../../api/httpClient';
 import { WaybillTableProps } from '../../common/interfaces_types';
+import Search from '../Searh';
 
 const WaybillTable: React.FC<WaybillTableProps> = (props: WaybillTableProps) => {
   const {
     waybills, setWaybillModalActive, setWaybillID, setCheckpoints,
   } = props;
 
+  const [searData, setSearData] = React.useState();
   const handleGetCheckpoint = (id) => {
     setWaybillModalActive(true);
     setWaybillID(id);
     httpClient.route.get_routes(id).then((response) => setCheckpoints(response.data));
   };
-
+  const data = searData || waybills;
   return (
     <div>
+      <Search setData={setSearData} Data={waybills} searchField="status" />
```

Comment:
From my point of view, searching by status is not "search", its just filtering. Maybe we need to investigate a bit more to find how can we differ waybills. Maybe we can show Consignment seria and number of each waybill and search this way. I suggest to discuss it with Sasha.